packages/flutter/lib/src/painting/gradient.dart

```diff
@@ -124,7 +125,41 @@ abstract class Gradient {
   /// If the gradient's configuration is text-direction-dependent, for example
   /// it uses [AlignmentDirectional] objects instead of [Alignment]
   /// objects, then the `textDirection` argument must not be null.
-  Shader createShader(Rect rect, { TextDirection textDirection });
+  ///
+  /// The `transform` property will be applied as a local transform to the
+  /// gradient, allowing the shader to be transformed without transforming the
+  /// entire canvas.
+  ///
+  /// {@tool sample}
+  /// This example shows how to rotate a [SweepGradient] without rotating the
+  /// [Canvas] or the [Rect] that the gradient is shading.
+  ///
+  /// ```dart
+  /// import 'dart:math' as math;
+  ///
+  /// void paint(Canvas canvas, Rect rect) {
+  ///   // Calculate 45 degrees (clockwise rotation) as radians.
+  ///   final double radians = 45 * math.pi / 180;
+  ///
+  ///   // Calculate the point to rotate about.
+  ///   final double sinRadians = math.sin(radians);
+  ///   final double oneMinusCosRadians = 1 - math.cos(radians);
+  ///   final Offset center = rect.center;
+  ///   final double originX = sinRadians * center.dy + oneMinusCosRadians * center.dx;
+  ///   final double originY = -sinRadians * center.dx + oneMinusCosRadians * center.dy;
+  ///
+  ///   final Matrix4 transform = Matrix4.identity()
+  ///     ..translate(originX, originY)
+  ///     ..rotateZ(radians);
```

Comment:
I suspect having amethod like this in the framework might be helpful to people, but I'm not sure where it would go.  MatrixUtils?

@goderbauer wdyt?